From 033ef1e3acf862f2bc7d4d2581599e514d6fbb4d Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Sun, 11 Aug 2019 20:35:50 +0200
Subject: [PATCH] Revert "xfs: report AG health via AG geometry ioctl"

This reverts commit 378d9e6d3edf4278e64864dc4ba0086f8a48b3af.

That commit leads to build failures:
../libxfs/.libs/libxfs.so: undefined reference to `xfs_ag_geom_health'"

https://marc.info/?l=linux-xfs&m=156552880421301&w=2

Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 libxfs/xfs_ag.c     |  2 --
 libxfs/xfs_fs.h     | 14 +-------------
 libxfs/xfs_health.h |  1 -
 3 files changed, 1 insertion(+), 16 deletions(-)

diff --git a/libxfs/xfs_ag.c b/libxfs/xfs_ag.c
index 084205b8..5fed507e 100644
--- a/libxfs/xfs_ag.c
+++ b/libxfs/xfs_ag.c
@@ -21,7 +21,6 @@
 #include "xfs_rmap.h"
 #include "xfs_ag.h"
 #include "xfs_ag_resv.h"
-#include "xfs_health.h"
 
 static struct xfs_buf *
 xfs_get_aghdr_buf(
@@ -507,7 +506,6 @@ xfs_ag_get_geometry(
 		   pag->pagf_btreeblks -
 		   xfs_ag_resv_needed(pag, XFS_AG_RESV_NONE);
 	ageo->ag_freeblks = freeblks;
-	xfs_ag_geom_health(pag, ageo);
 
 	/* Release resources. */
 	xfs_perag_put(pag);
diff --git a/libxfs/xfs_fs.h b/libxfs/xfs_fs.h
index f1158a79..486f56d4 100644
--- a/libxfs/xfs_fs.h
+++ b/libxfs/xfs_fs.h
@@ -306,21 +306,9 @@ struct xfs_ag_geometry {
 	uint32_t	ag_freeblks;	/* o: free space */
 	uint32_t	ag_icount;	/* o: inodes allocated */
 	uint32_t	ag_ifree;	/* o: inodes free */
-	uint32_t	ag_sick;	/* o: sick things in ag */
-	uint32_t	ag_checked;	/* o: checked metadata in ag */
 	uint32_t	ag_reserved32;	/* o: zero */
-	uint64_t	ag_reserved[12];/* o: zero */
+	uint64_t	ag_reserved[13];/* o: zero */
 };
-#define XFS_AG_GEOM_SICK_SB	(1 << 0)  /* superblock */
-#define XFS_AG_GEOM_SICK_AGF	(1 << 1)  /* AGF header */
-#define XFS_AG_GEOM_SICK_AGFL	(1 << 2)  /* AGFL header */
-#define XFS_AG_GEOM_SICK_AGI	(1 << 3)  /* AGI header */
-#define XFS_AG_GEOM_SICK_BNOBT	(1 << 4)  /* free space by block */
-#define XFS_AG_GEOM_SICK_CNTBT	(1 << 5)  /* free space by length */
-#define XFS_AG_GEOM_SICK_INOBT	(1 << 6)  /* inode index */
-#define XFS_AG_GEOM_SICK_FINOBT	(1 << 7)  /* free inode index */
-#define XFS_AG_GEOM_SICK_RMAPBT	(1 << 8)  /* reverse mappings */
-#define XFS_AG_GEOM_SICK_REFCNTBT (1 << 9)  /* reference counts */
 
 /*
  * Structures for XFS_IOC_FSGROWFSDATA, XFS_IOC_FSGROWFSLOG & XFS_IOC_FSGROWFSRT
diff --git a/libxfs/xfs_health.h b/libxfs/xfs_health.h
index 49ddfeac..8bf685f4 100644
--- a/libxfs/xfs_health.h
+++ b/libxfs/xfs_health.h
@@ -184,7 +184,6 @@ xfs_inode_is_healthy(struct xfs_inode *ip)
 }
 
 void xfs_fsop_geom_health(struct xfs_mount *mp, struct xfs_fsop_geom *geo);
-void xfs_ag_geom_health(struct xfs_perag *pag, struct xfs_ag_geometry *ageo);
 void xfs_bulkstat_health(struct xfs_inode *ip, struct xfs_bstat *bs);
 
 #endif	/* __XFS_HEALTH_H__ */
-- 
2.23.0.rc1

